testng-core/src/main/java/org/testng/internal/TestListenerHelper.java

```diff
@@ -35,7 +36,9 @@ public static void runPreConfigurationListeners(
 
   public static void runPostConfigurationListeners(
       ITestResult tr, ITestNGMethod tm, List<IConfigurationListener> listeners) {
-    for (IConfigurationListener icl : listeners) {
+    List<IConfigurationListener> listenersreversed = Lists.newArrayList(listeners);
+    Collections.reverse(listenersreversed);
+    for (IConfigurationListener icl : listenersreversed) {
```

Comment:
Here as well, we can use the newly added method `Lists.newReversedArrayList()`

Reply:
Fixed